src/Agent.Worker/Handlers/PowerShell3Handler.cs

```diff
@@ -73,6 +73,17 @@ public async Task RunAsync()
                                         killProcessOnCancel: false,
                                         inheritConsoleHandler: !ExecutionContext.Variables.Retain_Default_Encoding,
                                         cancellationToken: ExecutionContext.CancellationToken);
+
+            await System.Threading.Tasks.Task.WhenAny(step, ExecutionContext.ForceCompleted);
```

Comment:
thx, i'm guessing this is the same pattern as the node handler? if so this looks good to me, to merge.

Reply:
Correct, its mostly a copy paste of that code. I thought about refactoring it to be in a shared location, but I didn't come up with a great way to do that.